### PR TITLE
Project path env

### LINKED ISSUE #151: Prefer configuring project path via environment variables over buildpack.yml
## Summary
The current implementation allows a buildpack user to configure a project path using a `buildpack.yml` configuration file with the following contents:

```yaml
dotnet-build:
  project-path: ./src/asp_web_app
```

After the approval of [RFC0002](https://github.com/paketo-buildpacks/dotnet-core/blob/main/rfcs/0002-buildpack-yml-to-env-var.md) we should be preferring to allow that configuration via an environment variable like:

```shell
BP_DOTNET_PROJECT_PATH=./src/asp_web_app
```

### Deprecation Notice
The buildpack should emit a deprecation notice via log output when a `buildpack.yml` file appears in the application source code with configuration for this buildpack. The deprecation notice should appear similar to notices that have been implemented in [other](https://github.com/paketo-buildpacks/mri/blob/4e486427aa4ca316016403686ca1755cab9f554c/build.go#L68-L73) [buildpacks](https://github.com/paketo-buildpacks/go-dist/blob/de73c417686ab36e7686d881e2d0c769939f8ea5/build.go#L53-L58).

### PR DESCRIPTION
**Pull request by thitch97**
## Summary

Adds support for setting a custom project path via $BP_DOTNET_PROJECT_PATH

Resolves #151 

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

**Comment by emmjohnson** [comment]
Paketo doc changes - https://github.com/paketo-buildpacks/paketo-website/pull/128

**Comment by sophiewigmore** [comment]
@thitch97 @emmjohnson is this ready for review? What is it blocked on?

File: README.md
Comment (fg-j) [inline README.md:27]: It'd probably be useful to mention in the documentation that `BP_DOTNET_PROJECT_PATH` specifies the path to your source code app's cs/fs/vbproj file, specifically. Because in some cases, this buildpack runs on FDDs or FDEs where there is no project file _and in those cases, the environment variable has no effect_.

File: detect.go
Comment (fg-j) [inline detect.go:47]: Given that logging during detection is a) silenced by default by `pack` and b) prints in a nondeterministic order based on all the detecting buildpacks running in parallel, I think it'd be a bit clearer to log this warning during the build phase.

Granted, it's a bit weird to parse the `buildpack.yml` during build just to log a warning, but I think that'll provide a clearer UX and doesn't carry a significant performance penalty.

**Review by fg-j — CHANGES_REQUESTED** [review]
@sophiewigmore and I had a concern about logging warnings in detect rather than build. (See other comment). Otherwise, this seems to be working well.

File: build.go
Comment (thitch97) [inline build.go:20]: @paketo-buildpacks/dotnet-core-maintainers I don't think it's appropriate to error out here since this parser is only used to display the deprecation warning. On the other hand, the warning may be just as confusing for users. Any preference?

Comment (fg-j) [reply build.go:20]: I think it's probably fine to error here rather than warn. Hitting this case IRL is quite unlikely. I'd expect this call to fail in detect and/or for another buildpack in the order group.

I'd prefer to error out to avoid adding to this buildpack the eccentricity of warning in a failure case.

File: detect.go
Comment (sophiewigmore) [inline detect.go:32]: Is the logger in detect still necessary if we aren't logging the warning here anymore?

**Comment by sophiewigmore** [comment]
Besides that last small comment, this looks good to me. Since @fg-j has been looking at this PR I'll let her take a look as well before we approve!

**Review by fg-j — APPROVED** [review]

**Pull request merged**
